# 📘 Homework Pandas – Phân Tích Dữ Liệu

## Câu 1: Tạo Pandas Series

Tạo một `Series` chứa các giá trị: `1, 3, 5, None, 9`.

- In ra Series
- Kiểm tra kiểu dữ liệu (`dtype`)


In [24]:
# TODO
import pandas as pd
df=pd.Series([1, 3, 5, None, 9])
df


0    1.0
1    3.0
2    5.0
3    NaN
4    9.0
dtype: float64

## Câu 2: Tạo DataFrame từ Dictionary

Tạo một `DataFrame` từ dictionary sau:

```python
{
  'Name': ['Tom', 'Jerry', 'Mickey'],
  'Age': [20, 21, None],
  'City': ['NY', 'LA', 'FL']
}
```


In [25]:
# TODO
df=pd.DataFrame({
  'Name': ['Tom', 'Jerry', 'Mickey'],
  'Age': [20, 21, None],
  'City': ['NY', 'LA', 'FL']
})


## Câu 3: Truy xuất dữ liệu với loc & iloc

- Lấy thông tin hàng đầu tiên bằng `.iloc`
- Lấy cột `Age` bằng `.loc`


In [26]:
# TODO
first_row = df.iloc[0]
age_column = df.loc[:, 'Age']
print(first_row)
print(age_column)


Name     Tom
Age     20.0
City      NY
Name: 0, dtype: object
0    20.0
1    21.0
2     NaN
Name: Age, dtype: float64


## Câu 4: Import dữ liệu CSV
- Tải file `Titanic-Dataset.csv` từ đường link này : https://www.kaggle.com/datasets/yasserh/titanic-dataset
- Dùng `read_csv()` để đọc file `Titanic-Dataset.csv`
- Hiển thị 5 dòng đầu tiên
- In ra kích thước DataFrame


In [27]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("yasserh/titanic-dataset")

print("Path to dataset files:", path)



Using Colab cache for faster access to the 'titanic-dataset' dataset.
Path to dataset files: /kaggle/input/titanic-dataset


In [28]:
df=pd.read_csv('/kaggle/input/titanic-dataset/Titanic-Dataset.csv')
df.head()


PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [29]:
df.size

10692

## Câu 5: Khám phá dữ liệu ban đầu

- Dùng `info()` để kiểm tra kiểu dữ liệu
- Dùng `isnull().sum()` để đếm giá trị thiếu


In [30]:
# TODO
df.info()
df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

## Câu 6: Xử lý Missing Values

- Điền giá trị thiếu của cột `Age` bằng **giá trị trung bình**
- Kiểm tra lại số lượng missing values


In [31]:
# TODO
df['Age'].fillna(df['Age'].mean(), inplace=True)
df.isnull().sum()


/tmp/ipython-input-2467613277.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Age'].fillna(df['Age'].mean(), inplace=True)


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

## Câu 7: Xử lý dữ liệu trùng lặp

- Kiểm tra số dòng bị trùng
- Xóa các dòng trùng lặp (nếu có)


In [32]:
# TODO
df[df.duplicated()]


Empty DataFrame
Columns: [PassengerId, Survived, Pclass, Name, Sex, Age, SibSp, Parch, Ticket, Fare, Cabin, Embarked]
Index: []

## Câu 8: Feature Selection

Chỉ giữ lại các cột sau trong Titanic dataset:
`Survived`, `Pclass`, `Sex`, `Age`, `Fare`


In [33]:
# TODO
df.drop(columns=['PassengerId', 'Name', 'SibSp', 'Parch', 'Ticket', 'Cabin', 'Embarked'], inplace=True)
df

Survived  Pclass     Sex        Age     Fare
0           0       3    male  22.000000   7.2500
1           1       1  female  38.000000  71.2833
2           1       3  female  26.000000   7.9250
3           1       1  female  35.000000  53.1000
4           0       3    male  35.000000   8.0500
..        ...     ...     ...        ...      ...
886         0       2    male  27.000000  13.0000
887         1       1  female  19.000000  30.0000
888         0       3  female  29.699118  23.4500
889         1       1    male  26.000000  30.0000
890         0       3    male  32.000000   7.7500

[891 rows x 5 columns]

## Câu 9: Thống kê mô tả

- Dùng `describe()` cho các cột số
- Tính tuổi trung bình của hành khách


In [34]:
# TODO
print(df.describe())
print(df['Age'].mean())

         Survived      Pclass         Age        Fare
count  891.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118   32.204208
std      0.486592    0.836071   13.002015   49.693429
min      0.000000    1.000000    0.420000    0.000000
25%      0.000000    2.000000   22.000000    7.910400
50%      0.000000    3.000000   29.699118   14.454200
75%      1.000000    3.000000   35.000000   31.000000
max      1.000000    3.000000   80.000000  512.329200
29.69911764705882


## Câu 10: Mini Insight (Tự luận)

Dựa trên phân tích Titanic dataset, hãy trả lời:

1. Giới tính nào có tỷ lệ sống sót cao hơn?
2. Hạng vé (`Pclass`) ảnh hưởng như thế nào đến khả năng sống sót?

**Trả lời bằng Markdown cell (3–5 dòng).**


In [37]:
print(df.groupby('Sex')['Survived'].mean())
print(df.groupby('Pclass')['Survived'].mean())

Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64
Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


*Viết câu trả lời của bạn tại đây*

> 1. Group 'Sex' và 'Survived' và tính mean ta thấy nữ có tỉ lệ sống sót cao hơn(0.742038=~ 74%) so với nam(chỉ 0.188908=~ 19%)
2. Dùng tương tự cách trên với 'Pclass' và 'Survived' có thể thấy được hạng vé càng "VIP" thì khả năng sống sót càng cao(1>2>3)

